### ISSUE TITLE
Add support for Python 3.12

### ISSUE THREAD

**Issue by vkottler** [vkottler/vcorelib#130]
Blocked on: https://github.com/MagicStack/uvloop/issues/567.

`vmklib` is ready: https://github.com/vkottler/vmklib/pull/80.

**Comment by vkottler** [comment]
WIP on this PR: https://github.com/vkottler/vcorelib/pull/129.

**Comment by vkottler** [comment]
![image](https://github.com/vkottler/vcorelib/assets/15205052/2b1c9c6f-a6c4-4f63-820d-98e702efff3d)

we could also add an interpreter version constraint to this potentially

**Comment by vkottler** [comment]
New issue now: https://github.com/MagicStack/uvloop/issues/575, can work around it though.

**Comment by vkottler** [comment]
This is merged.
